packages/hawtio/src/plugins/springboot/Health.tsx

```diff
@@ -118,33 +103,53 @@ export const Health: React.FunctionComponent = () => {
               </CardHeader>
             </Card>
           </GridItem>
-          {healthData?.components.map(component => (
-            <GridItem
-              span={componentSpanRecord[component.name]![0] as gridSpans}
-              key={component.name}
-              mdRowSpan={componentSpanRecord[component.name]![1] as gridSpans}
-            >
-              <Card isFullHeight>
-                <CardHeader>
-                  <Title headingLevel='h3'>{humanizeLabels(component.name!)}</Title>
-                </CardHeader>
-                <CardBody>
-                  <Flex>
-                    <FlexItem>
-                      <HealthStatusIcon status={component.status} />
-                    </FlexItem>
-                    <FlexItem>Status: {component.status}</FlexItem>
-                    {component.details &&
-                      (component.name === 'diskSpace' ? (
-                        <DiskComponentDetails componentDetails={component.details} />
-                      ) : (
-                        <ComponentDetails componentDetails={component.details} />
-                      ))}
-                  </Flex>
-                </CardBody>
-              </Card>
-            </GridItem>
-          ))}
+          {healthData?.components.map(
+            component =>
+              HEALTH_COMPONENTS.includes(component.name) && (
```

Comment:
You map everything here and then effectively use an `if` condition to decide whether display the component. However, underneath you filter and them map. Would it be better to use a uniform method across both or do a single map of all components then pick one component choice or other depending on the `includes` ?

Reply:
yeah I know what do you mean. This is more-less a substitute for a simple sorting e.g health and camel on top with fixed span and rest on bottom but I will try to sort them.